#! /bin/sh /usr/share/dpatch/dpatch-run
## fix-lpstat.dpatch by Evan Broder <broder@mit.edu>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix a bug in CUPS 1.3's lpstat talking to 1.4 servers.
## 

@DPATCH@
diff -urNad cupsys-1.3.7~/systemv/lpstat.c cupsys-1.3.7/systemv/lpstat.c
--- cupsys-1.3.7~/systemv/lpstat.c	2007-07-11 17:46:42.000000000 -0400
+++ cupsys-1.3.7/systemv/lpstat.c	2009-12-16 15:12:09.000000000 -0500
@@ -1433,11 +1433,15 @@
   *    attributes-charset
   *    attributes-natural-language
   *    job-uri
+  *    printer-uri
   *    requested-attributes
   */
 
   request = ippNewRequest(IPP_GET_JOBS);
 
+  ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_URI, "printer-uri",
+               NULL, "ipp://localhost/");
+
   ippAddStrings(request, IPP_TAG_OPERATION, IPP_TAG_KEYWORD,
                 "requested-attributes", sizeof(jattrs) / sizeof(jattrs[0]),
 		NULL, jattrs);
